### PR TITLE
fix: tighten overly loose body patterns to prevent false positives

### PR DESCRIPTION
**Pull request by royisy**
## Summary

- Scope the `ant_design` and `tailwind_css` body patterns to `<link>` tag attribute values so they no longer match unrelated `href` + keyword co-occurrences elsewhere in the document (e.g. case-insensitive substring matches in minified JS, or anchor tags that link to pages *about* the library).
- Use `\bantd\b` word boundaries in the ant_design CSS pattern so common filenames like `antd.css` / `antd.min.css` are still detected while sub-word matches (e.g. `antdate.css`, `merchantDashboard.css`) are rejected.
- Require a version number in the `swiper` body pattern — previously the optional group let `Swiper ` alone match any prose that mentioned the product.
- Add unit tests for each signature, including negative cases for the specific false-positive shapes the old patterns allowed.

## Test plan

- [x] `npx vitest run src/signatures` — all signature tests pass (169/169)
- [x] ant_design: positive cases for `ant-*` class, `anticon`, `antd@X.Y.Z` CDN, `antd.css`, `antd.min.css`
- [x] ant_design: negative cases for `antdate.css`, `merchantData`-style sub-word, minified JS with incidental `antD`
- [x] tailwind_css: existing positive cases still pass; new negative case for `<a href="/blog/.../tailwind-tips">`
- [x] swiper: version banner detection, CDN URL detection, prose-mention FP guard

### CODE REVIEW

File: src/signatures/technologies/ant_design.ts [hunk 1]
```diff
@@ -2,14 +2,14 @@ import type { Signature } from "../_types.js";
...
   rule: {
     confidence: "high",
     bodies: [
-      "<[^>]*class=\"ant-(?:btn|col|row|layout|breadcrumb|menu|pagination|steps|select|cascader|checkbox|calendar|form|input-number|input|mention|rate|radio|slider|switch|tree-select|time-picker|transfer|upload|avatar|badge|card|carousel|collapse|list|popover|tooltip|table|tabs|tag|timeline|tree|alert|modal|message|notification|progress|popconfirm|spin|anchor|back-top|divider|drawer)",
-      "<i class=\"anticon anticon-",
-      "href[^>]+antd(?:@|/)?([\\d\\.]+)?(?:.+|)?\\.css",
-      "href[^>]+antd",
+      '<[^>]*class="ant-(?:btn|col|row|layout|breadcrumb|menu|pagination|steps|select|cascader|checkbox|calendar|form|input-number|input|mention|rate|radio|slider|switch|tree-select|time-picker|transfer|upload|avatar|badge|card|carousel|collapse|list|popover|tooltip|table|tabs|tag|timeline|tree|alert|modal|message|notification|progress|popconfirm|spin|anchor|back-top|divider|drawer)',
+      '<i class="anticon anticon-',
+      "<link[^>]+href=[\"'][^\"']*\\bantd\\b(?:@([\\d\\.]+))?[^\"']*\\.css",
```
[the remark this answers is not in the document]
Comment (royisy) [reply src/signatures/technologies/ant_design.ts]: Fixed

File: src/signatures/technologies/ant_design.test.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+      const result = applySignature(context, antDesignSignature);
+      expect(result).toBeUndefined();
+    });
+
+    it("does not detect Ant Design from an unrelated stylesheet whose filename contains 'antd' as a sub-word", () => {
+      const context = createMockContext({
+        responses: [
+          createMockResponse({
+            body: '<link rel="stylesheet" href="/assets/relevantDashboard.css">',
```
[the remark this answers is not in the document]
Comment (royisy) [reply src/signatures/technologies/ant_design.test.ts]: Fixed

**Review by 0n1shi — APPROVED** [review]
